addon/object/index.js

```diff
@@ -147,11 +147,11 @@ export const action = decorator(function(target, key, desc) {
  * @param {...String} propertyNames - List of property keys this computed is dependent on
  */
 export const computed = decoratorWithParams(function(target, key, desc, params) {
-  if (!desc.writable) {
-    throw new Error('ember-decorators does not support using getters and setters');
+  if (desc.value instanceof Ember.ComputedProperty) {
```

Comment:
Open to bikeshedding over the correct behavior here. If a user tries to decorate _both_ the getter and setter, the decorator will run twice on the same descriptor (the first time it already has both the getter and setter), so currently we throw an error. 

However, unlike the current `get`/`set` syntax which basically _requires_ both functions to be side-by-side, setters and getters can be anywhere in a class definition. So it may make sense to allow annotation on both to prevent confusion (e.g. this setter _is_ part of a computed, this one is a plain ES setter)